File: frontend/src/services/categoryService.js
import api from './api';

const CATEGORY_ENDPOINTS = {
  BASE: '/v1/categories',
  BY_ID: (id) => `/v1/categories/${id}`,
  BY_SLUG: (slug) => `/v1/categories/slug/${slug}`,
  TREE: '/v1/categories/tree',
  FILTERS: '/v1/categories/filters',
};

export const categoryService = {
  getAll: () => api.get(CATEGORY_ENDPOINTS.BASE),

  getById: (id) => api.get(CATEGORY_ENDPOINTS.BY_ID(id)),

  getBySlug: (slug) => api.get(CATEGORY_ENDPOINTS.BY_SLUG(slug)),

  getTree: () => api.get(CATEGORY_ENDPOINTS.TREE),

  getFilters: () => api.get(CATEGORY_ENDPOINTS.FILTERS),

  create: (data) => api.post(CATEGORY_ENDPOINTS.BASE, data),

  update: (id, data) => api.put(CATEGORY_ENDPOINTS.BY_ID(id), data),

  delete: (id) => api.delete(CATEGORY_ENDPOINTS.BY_ID(id)),
};

File: backend/product-service/src/main/java/com/demir/ecommerce/productservice/service/CategoryService.java
package com.demir.ecommerce.productservice.service;

import com.demir.ecommerce.productservice.dto.category.request.CategoryCreateRequest;
import com.demir.ecommerce.productservice.dto.category.response.CategoryFilterResponse;
import com.demir.ecommerce.productservice.dto.category.response.CategoryResponse;
import com.demir.ecommerce.productservice.dto.category.response.CategoryTreeResponse;
import com.demir.ecommerce.productservice.dto.category.request.CategoryUpdateRequest;

import java.util.List;

public interface CategoryService {

    CategoryResponse create(CategoryCreateRequest request);

    CategoryResponse update(Long id, CategoryUpdateRequest request);

    CategoryResponse getById(Long id);

    CategoryResponse getBySlug(String slug);

    List<CategoryResponse> getAll();

    List<CategoryTreeResponse> getTree();

    List<CategoryFilterResponse> getFilters();

    void delete(Long id);
}